src/shapes/IText/ITextBehavior.ts

```diff
@@ -387,6 +387,22 @@ export abstract class ITextBehavior<
     if (this.isEditing || !this.editable) {
       return;
     }
+    this._enterEditing();
+    this.fire('editing:entered', e ? { e } : undefined);
+    this._fireSelectionChanged();
+    if (this.canvas) {
+      this.canvas.fire('text:editing:entered', {
+        target: this as unknown as IText,
+        e,
+      });
+      this.canvas.requestRenderAll();
+    }
+  }
+
+  /**
+   * runs the actual logic that enter from editing state, see {@link enterEditing}
+   */
+  protected _enterEditing() {
```

Comment:
if is protected you can't really call it by yourself. This should stay public and have a different name.
Maybe enterEditingImpl ( as for implementation ) is better than the underscore prefix?

Reply:
I found that other internal methods that do not trigger the fire event are all prefixed with an underscore, such as 
 _discardActiveObject/discardActiveObject and _exitEditing/exitEditing, which is why I named it this way.
I should also change the _exitEditing method to public.

Reply:
To unify, I changed the method name _exitEditing to exitEditingImpl.